### PR TITLE
Carrie Walsh

### PR DESCRIPTION
**Pull request by carriewalsh**
Week 5 - here it is!

### CODE REVIEW

File: lib/market.rb [hunk 1]
```diff
@@ -1,3 +1,55 @@
...
+    @vendors << vendor
+  end
+
+  def vendor_names
+    @vendors.map {|vendor| vendor.name}
+  end
+
+  def vendors_that_sell(item)
+    @vendors.find_all do |vendor|
+      vendor.inventory.has_key?(item)
```
Comment (ameseee) [inline lib/market.rb]: Awesome!

File: lib/market.rb [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+    end
+  end
+
+  def sorted_item_list
+    list = @vendors.map {|vendor| vendor.inventory.keys}
+    list.flatten!.uniq!.sort!
+  end
+
+  def total_inventory
+    hash = Hash.new {|hash,key| hash[key] = 0}
```
Comment (ameseee) [inline lib/market.rb]: Nice job! I'd suggest using a more descriptive name than `hash` for this variable in the future😊

File: test/vendor_test.rb [hunk 3]
```diff
@@ -3,19 +3,36 @@
...
+  def test_stocking_items_adds_to_inventory
+    assert_equal 30, @vendor.inventory["Peaches"]
+    @vendor.stock("Tomatoes",12)
+    expected = {"Peaches" => 30,
+                "Tomatoes" => 12}
+    assert_equal expected, @vendor.inventory
+  end
+
+  def test_vendor_can_check_stock_of_item
+    assert_equal 30, @vendor.check_stock("Peaches")
```
Comment (ameseee) [inline test/vendor_test.rb]: If you stocked 20 more peaches, are you sure that the stock would not be at 50? Or would it override to 20? 😬

File: lib/vendor.rb [hunk 4]
```diff
@@ -4,7 +4,14 @@ class Vendor
 
   def initialize(name)
     @name = name
-    @inventory = # you decide what goes here...
+    @inventory = {}
   end
 
+  def stock(name,quantity)
+    @inventory[name] = quantity
```
Comment (ameseee) [inline lib/vendor.rb]: I commented on your test for this - be careful! What you currently have looks like it will override if you were to call:

```
vendor.stock("cheese", 10)
vender.stock("cheese", 15)
```

So it would end up at 15 instead of 25. This is where those more edge-case-like tests are super important. 

Let me know if that isn't super clear or you want to talk about more:)

File: test/market_test.rb [hunk 5]
```diff
@@ -2,7 +2,81 @@
...
+  def test_market_has_name
+    assert_equal "South Pearl Street Farmers Market", @market.name
+  end
+
+  def test_market_starts_with_no_vendors_by_default
+    market = Market.new("7-11")
+    assert_equal [], market.vendors
+  end
+
+  def test_market_can_have_vendors_added
```
Comment (ameseee) [inline test/market_test.rb]: Would love to see a test that verifies you can't ad the same vendor multiple times!

File: lib/market.rb [hunk 6]
```diff
@@ -1,3 +1,55 @@
...
+    @vendors.each do |vendor|
+      vendor.inventory.each do |item|
+        hash[item.first] += item.last
+      end
+    end
+    hash
+  end
+
+  def sell(item,quantity)
```
Comment (ameseee) [inline lib/market.rb]: Would love to find a way to refactor as we want to keep methods around that 8 line limit! 

Could pull that each starting on line 40 out of there as a start.

**Comment by ameseee** [comment]
Hi @carriewalsh - nice job overall. You're still in a good place but I left a couple comments to push you (one on an actual issues, one just to go above and beyond😉) Keep it up.
